activesupport/lib/active_support/message_verifier.rb

```diff
@@ -106,12 +115,26 @@ def verify(signed_message)
     #
     #   verifier = ActiveSupport::MessageVerifier.new 's3Krit'
     #   verifier.generate 'a private message' # => "BAhJIhRwcml2YXRlLW1lc3NhZ2UGOgZFVA==--e2d724331ebdee96a10fb99b089508d1c72bd772"
-    def generate(value)
-      data = encode(@serializer.dump(value))
-      "#{data}--#{generate_digest(data)}"
+
+    def generate(options)
```

Comment:
Yes, that’s closer. But I’d rather we go with `payload` instead of `value` in: `Claims.new(payload: value, options)`

> Den 28/06/2015 kl. 20.42 skrev Siddharth Bhatore <notifications@github.com>:
> 
> In activesupport/lib/active_support/message_verifier.rb <https://github.com/sbhatore/rails/pull/1#discussion_r33425555>:
> 
> > @@ -106,12 +115,26 @@ def verify(signed_message)
> >      #
> >      #   verifier = ActiveSupport::MessageVerifier.new 's3Krit'
> >      #   verifier.generate 'a private message' # => "BAhJIhRwcml2YXRlLW1lc3NhZ2UGOgZFVA==--e2d724331ebdee96a10fb99b089508d1c72bd772"
> > -    def generate(value)
> > -      data = encode(@serializer.dump(value))
> > -      "#{data}--#{generate_digest(data)}"
> > +
> > +    def generate(options)
> So, would it be better if I do it like this ?
> 
> def generate(value, **options)
>   @claims = Claims.new(value: value, **options)
>   data = [encode(serialized_header), encode(serialized_claims)].join('.')
>   "#{data}.#{generate_digest(data)}"
> end
> —
> Reply to this email directly or view it on GitHub <https://github.com/sbhatore/rails/pull/1/files#r33425555>.
> 

--
Kasper